config/fcos/v1_5_exp/translate.go

```diff
@@ -292,3 +302,55 @@ func translateBootDeviceLuks(from BootDeviceLuks, options common.TranslateOption
 	tm.AddTranslation(path.New("yaml"), path.New("json"))
 	return
 }
+
+func (c Config) processPackages(options common.TranslateOptions) (types.Config, report.Report, translate.TranslationSet) {
+	yamlPath := path.New("yaml", "extensions")
+
+	ret := types.Config{}
+	ts := translate.NewTranslationSet("yaml", "json")
+	var r report.Report
+
+	if len(c.Extentions) == 0 {
+		return ret, r, ts
+	}
+
+	type TreeFile struct {
+		Packages []string `yaml:"packages"`
+	}
+
+	p := TreeFile{
+		Packages: c.Extentions,
+	}
+
+	marshalledYAML, err := yaml.Marshal(&p)
+
+	if err != nil {
+		r.AddOnError(yamlPath, err)
+		return ret, r, ts
+	}
+
+	treeFilePath := "/etc/rpm-ostree-layer.yaml"
+
+	file := types.File{
+		Node: types.Node{
+			Path: treeFilePath,
+		},
+	}
+
+	src, gzipped, err := baseutil.MakeDataURL([]byte(marshalledYAML), file.Contents.Compression, !options.NoResourceAutoCompression)
+	if err != nil {
+		r.AddOnError(yamlPath, err)
+		return ret, r, ts
+	}
+	if gzipped {
+		file.Contents.Compression = util.StrToPtr("gzip")
+	}
+	file.Contents.Source = util.StrToPtr(src)
+
+	file.Mode = util.IntToPtr(0644)
+
+	ret.Storage.Files = append(ret.Storage.Files, file)
+	ts.AddFromCommonSource(yamlPath, path.New("json", "storage", "files", 0), file)
```

Comment:
This fails to add paths for `storage` and `storage.files`.
```suggestion
	ts.AddFromCommonSource(yamlPath, path.New("json", "storage"), ret.Storage)
```